crates/fluvio/src/config/tls.rs

```diff
@@ -241,13 +241,29 @@ cfg_if::cfg_if! {
                 }
             }
         }
-    }  else if #[cfg(feature = "rustls")] {
+    } else if #[cfg(any(feature = "rustls"))] {
 
         impl TryFrom<TlsPolicy> for DomainConnector {
             type Error = anyhow::Error;
 
             fn try_from(config: TlsPolicy) -> Result<Self, Self::Error> {
 
+                if rustls::crypto::CryptoProvider::get_default().is_none() {
```

Comment:
Since fluvio is library,  not sure this make sense.  If other binary decided to do change how this work, this will override and could cause conflict.  

I think it's better this to be done outside fluvio and move to `fluvio future` and make it available as API and part of initialization